### PR TITLE
fix: if prerelease, publish prerelease (python-semantic-release#544)

### PR DESCRIPTION
**Pull request by wintero92**
* Fixes publishing process of release/prerelease

Running `semantic-release publish --prerelease` currently creates release, not prerelease.

### CODE REVIEW

**Review by bernardcooke53 — COMMENTED** [review]
Looks good, I would be grateful to see a couple of additional tests before merging though:
* One in `test_hvcs.py` that checks when `create_release` is called with `prerelease=True`, the request body includes the correct value for `prerelease` similar to [this assertion](https://github.com/python-semantic-release/python-semantic-release/blob/a4b625f98da8b4d9364f6dd37260d92a24238cf6/tests/test_hvcs.py#L430) - even if that's just parameterising the existing test
* One in `test_cli.py` that ensures when `--prerelease` is supplied on the command line, `post_changelog` is called with the correct value for `prerelease`.

Thanks for looking at this!

**Comment by wintero92** [comment]
> Looks good, I would be grateful to see a couple of additional tests before merging though:
> 
> * One in `test_hvcs.py` that checks when `create_release` is called with `prerelease=True`, the request body includes the correct value for `prerelease` similar to [this assertion](https://github.com/python-semantic-release/python-semantic-release/blob/a4b625f98da8b4d9364f6dd37260d92a24238cf6/tests/test_hvcs.py#L430) - even if that's just parameterising the existing test
> * One in `test_cli.py` that ensures when `--prerelease` is supplied on the command line, `post_changelog` is called with the correct value for `prerelease`.
> 
> Thanks for looking at this!

You will probably want to check naming and position of the tests in respective files.

**Comment by wintero92** [comment]
Any change you had a time to look at this? @bernardcooke53

File: tests/test_hvcs.py [hunk 1]
```diff
@@ -686,7 +725,46 @@ def request_callback(request):
...
             with mock.patch.dict(os.environ, {"NETRC": netrc_file.name}):
-                status = Gitea.post_release_changelog("gitea", "tea", "1.0.0", "text")
+                status = Gitea.post_release_changelog(
+                    "gitea", "tea", "1.0.0", "text", False
+                )
+                self.assertTrue(status)
+
+    @responses.activate
+    @mock.patch("semantic_release.hvcs.Gitea.token", return_value=None)
+    def test_should_create_prerelease_if_asked_for(self, mock_token):
```
Comment (bernardcooke53) [inline tests/test_hvcs.py:735]: This is the duplicate I mentioned above

File: tests/test_hvcs.py [hunk 2]
```diff
@@ -445,7 +445,44 @@ def request_callback(request):
...
             with mock.patch.dict(os.environ, {"NETRC": netrc_file.name}):
                 status = Github.post_release_changelog(
-                    "relekang", "rmoq", "1.0.0", "text"
+                    "relekang", "rmoq", "1.0.0", "text", False
+                )
+                self.assertTrue(status)
+
+    @responses.activate
+    @mock.patch("semantic_release.hvcs.Github.token", return_value=None)
+    def test_should_create_prerelease_if_asked_for(self, mock_token):
```
Comment (bernardcooke53) [inline tests/test_hvcs.py:454]: I think you've duplicated the test name with the other flavours of git remotes - have left a comment below.
I don't think copy-paste is needed compared to the existing test cases - it should be fine to just add
```python
@pytest.mark.parametrize("prerelease", [True, False])
```
to the existing test along with the `prerelease` parameter, if that's easier (helps avoid repetition)

edit: forgot to click "save" when submitting the review

**Comment by xec-abailey** [comment]
Howdy! Any chance this will land at some point? would love to have proper labeling of pre-releases. Happy to contribute to get this pushed through as well if that's possible/needed

**Comment by tofuurem** [comment]
@bernardcooke53 I hope you can find the time to check it out (PR#587)

**Comment by bernardcooke53** [comment]
Thanks for the nudge @tofuurem , I've been thoroughly slammed for time in the last couple of weeks with my day job but going to devote some time to issues and PRs at the weekend - I'll go over it then👍

**Comment by bernardcooke53** [comment]
Merged #587 which supersedes this PR - thank you for the contribution @wintero92 :pray:

**Pull request closed without merging**